capstone/src/main/java/com/step902020/capstone/EventController.java

```diff
@@ -24,27 +27,53 @@
   @Autowired
   private EventRepository eventRepository;
 
-  @GetMapping("/list-events")
-  public Iterable<EventTemp> getEvents() {
+  @GetMapping("get-all-events")
+  public Iterable<Event> getAllEvents() {
+
     return this.eventRepository.findAll();
   }
 
-  @PostMapping("/new-event")
-  public void saveEvent() throws IOException {
-    this.eventRepository.save(new EventTemp("EVENT_HOLDER"));
+  @GetMapping("get-single-event")
+  public List<Event> getEventByID(@RequestParam("datastoreID") Long datastoreID) {
+    return this.eventRepository.findByDatastoreID(datastoreID);
   }
+//   @PostMapping("/new-event")
+//   public void saveEvent() throws IOException {
+//     this.eventRepository.save(new EventTemp("EVENT_HOLDER"));
+//   }
 
-  @PostMapping("/new-review")
-  public void addReview(
-      @RequestParam("text") String text,
-      @RequestParam("eventId") long eventId,
-      @RequestParam("name") String name) throws IOException {
+//   @PostMapping("/new-review")
+//   public void addReview(
+//       @RequestParam("text") String text,
+//       @RequestParam("eventId") long eventId,
+//       @RequestParam("name") String name) throws IOException {
  
-  EventTemp event =this.eventRepository.findById(eventId);
-  LocalDateTime date = LocalDateTime.now();
+//   EventTemp event =this.eventRepository.findById(eventId);
+//   LocalDateTime date = LocalDateTime.now();
 
-  event.addReview(new Review(date, text, name));
-  this.eventRepository.save(event);
-  }
+//   event.addReview(new Review(date, text, name));
+//   this.eventRepository.save(event);
+
+//   }
+
+
+  @PostMapping("save-event")
+  public RedirectView saveEvent (
+    @RequestParam("organizationName") String organizationName,
+    @RequestParam("eventTitle") String eventTitle,
+    @RequestParam("eventDateTime") String eventDateTime,
+    @RequestParam("eventDescription") String eventDescription,
+    @RequestParam("eventLatitude") String eventLatitude,
+    @RequestParam("eventLongitude") String eventLongitude
+    // @RequestParam("eventFilters") List<String> eventFilters
+    ) throws IOException {
+      
+      Event newEvent = new Event(organizationName, eventTitle, eventDateTime, eventDescription, Double.parseDouble(eventLatitude), Double.parseDouble(eventLongitude));
+
+      this.eventRepository.save(newEvent);
+      return new RedirectView("event.html", true);
```

Comment:
A possible enhancement (you can create a follow-up issue; it's not critical) is to redirect to that specific event.

Reply:
I will look into it!